java/core/src/main/java/com/microsoft/bond/protocol/StreamNotCloneableException.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond.protocol;
+
+import java.io.IOException;
+
+/**
+ * Indicates that a stream cannot be cloned because the original stream is not cloneable.
+ */
+public class StreamNotCloneableException extends IOException {
```

Comment:
I disagree. Stream cloning is an operation in input and hence an I/O operation. The error is that there is no support for the particular type of input. In my mind, it's not much different from UnsupportedEncodingException and other like that.

---
In reply to: [130514052](https://github.com/Microsoft/bond/pull/540#discussion_r130514052) [](ancestors = 130514052)